hatchery/nextflow.go

```diff
@@ -981,7 +988,8 @@ func setupSquid(hostname string, userName string, cidrstring string, ec2svc *ec2
 	subnetName := fmt.Sprintf("%s-nf-subnet-fw-%s", hostname, userName)
 	Config.Logger.Printf("Debug: Creating subnet '%s' with name '%s'", subnet, subnetName)
 
-	subnetId, err := setupSubnet(subnetName, subnetString, vpcid, ec2svc)
+	// TODO: read instance type from config. (MIDRC-751)
+	subnetId, err := setupSubnet(subnetName, subnetString, vpcid, ec2svc, "t2.micro")
```

Comment:
This hardcoded string will be changed to a configurable value as a part of a future ticket. Right?

Reply:
yes, see comment. `MIDRC-751`

Reply:
looking at the changes this doesn't look right: `setupSubnet`'s `instanceType` parameter is the type we want for nextflow jobs instances, it's not the squid instance type we want to configure in MIDRC-751 - it should be `g4dn.xlarge` not `t2.micro`. I think you got it confused with [this](https://github.com/uc-cdis/hatchery/blob/4d5493ebf144bcbe6f7541a014907e89a8f617cd/hatchery/nextflow.go#L1348)?

so i think it should be:
```
setupSubnet(... availableInstanceTypes=["g4dn.xlarge"])

[...]

// TODO: read instance type from config. (MIDRC-751)
launchSquidInstance(... instanceType="t2.micro")
```

Reply:
You are right pauline. This instance type is just indicating the value for the subnet. It is taking from config as expected. `t2.micro` is sent as an argument to `setupSubnet` only within the `setupSquid` method ([here](https://github.com/uc-cdis/hatchery/blob/15114cebe0c3c126dddc67c37cd40f01278d42b6/hatchery/nextflow.go#L1020C23-L1020C23)) whereas `setupSubnet` method while creating Nextflow Resources uses the instance type that is defined in the hatchery config ([here](https://github.com/uc-cdis/hatchery/blob/15114cebe0c3c126dddc67c37cd40f01278d42b6/hatchery/nextflow.go#L113C93-L113C93))

Reply:
@nss10 I pushed this change https://github.com/uc-cdis/hatchery/pull/111/commits/5542a4251e46978b2a08cc9624cdb5c3a8c0dd08 to pass the squid instance type as a parameter instead of hardcoding "t2.micro" twice. Would appreciate if you can check it and maybe test the branch again 🙏